### ISSUE TITLE
Implement EFS support

### ISSUE THREAD

**Issue by hackdna** [galaxyproject/cloudman#51]
Galaxy FS and indices EBS volumes on the master node are shared via NFS to the worker nodes and can become a bottleneck for IO intensive jobs.

[Amazon EFS](https://aws.amazon.com/efs/) is a network file storage service for Amazon EC2 instances that:
- supports the NFSv4 protocol
- scales throughput and IOPS as the file system grows
- grows and shrinks automatically as you add and remove files

Using EFS would increase cluster performance and reduce the processing and network load on the master node.

**Comment by nuwang** [comment]
Hi

Nfs mounts are supported in cloudman and by extension, I'd imagine that EFS would work ok. However the process is a bit manual at present. The process I usually follow is to add the nfs mount via the "add new filesystem" option in cloudman's admin page with the persistent option on.
Once mounted, the data in /mnt/galaxy can be synced across to the nfs.
Then, edit persistent-data.yml in the cluster bucket, remove the existing /mnt/galaxy filesystem and assign it to the nfs filesystem instead. The most important thing here is to copy the filesystem role attribute across.
An immediate reboot (not through cloudman) should cause the cluster to come up again with the nfs system as the galaxy filesystem.

**Comment by hackdna** [comment]
Thanks for the tip. Still, it would be great to be able to do this automatically.

**Comment by afgane** [comment]
Hmm, I looked at doing this natively and we'll have to wait a bit to see where it goes. For starters, we don't have access to EFS yet (I've applied). Even so, EFS is available only in the Oregon region, where Galaxy is not available currently so that poses a cost issue for us to host the necessary pieces. More problematic is that it seems that boto2 does not have support for EFS, which means we'd have to switch to boto3. With our aims to overhaul most of the backend of CloudMan to use CloudBridge, switch to boto3 now would be a very short-term effort and hence questionaly worth the investment. It may be possible to use boto3 just for this one piece of the puzzle but we'll have to see once we get access. I'll keep updating this issue as things move along.
